src/equations/ideal_glm_mhd_multicomponent_2d.jl

```diff
@@ -36,6 +36,23 @@ mutable struct IdealGlmMhdMulticomponentEquations2D{NVARS, NCOMP, RealT <: Real}
 
         new(gammas, gas_constants, cv, cp, c_h)
     end
+
+    # Inner constructor for `@reset` works correctly
+    function IdealGlmMhdMulticomponentEquations2D{NVARS, NCOMP, RealT}(gammas::SVector{NCOMP,
```

Comment:
This duplicates a lot of code. Can you please just modify the other inner constructor and adapt the outer constructors accordingly?

Reply:
I think this is just a temporary solution to address the mutability issue (would you prefer to treat it as final?) so that if we find a better approach, we can revert it more easily.

Reply:
Do you prefer this way to handle the mutability issue or one of the other drafts? If we go this way, I would like a cleaner approach for Trixi.jl.

Reply:
This way please - I really need move ahead.